### PR TITLE
Move pkg/registry/rbac code imported by kubectl under component helpers

### PR DESCRIPTION
**Pull request by ingvagabund**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Remove dependency of kubectl command on `k8s.io/kubernetes/pkg/registry/rbac/reconciliation` to get closer to moving kubectl out of the main repo.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

Building on top of https://github.com/liggitt/kubernetes/commits/extract-rbac

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by ingvagabund** [comment]
/hold
until dependency on `k8s.io/apiserver` is removed

**Comment by ingvagabund** [comment]
Dependency on k8s.io/apiserver is removed. The PR is too big to be merged. I will break the PR into two:
- moving `k8s.io/apiserver/pkg/authentication/serviceaccount` and `k8s.io/apiserver/pkg/authentication/user` under `k8s.io/component-helpers`
- moving `pkg/registry/rbac/validation` and `pkg/registry/rbac/reconciliation` under `k8s.io/component-helpers`

**Comment by ingvagabund** [comment]
PR https://github.com/kubernetes/kubernetes/pull/95793 for moving `k8s.io/apiserver/pkg/authentication/serviceaccount` and `k8s.io/apiserver/pkg/authentication/user` bits. Once merged, I will rebase this PR.

File: cmd/kubeadm/app/phases/bootstraptoken/clusterinfo/clusterinfo.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	"k8s.io/api/core/v1"
 	rbac "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/component-helpers/auth/user"
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/bootstraptoken/clusterinfo/clusterinfo.go:26]: /approve
for kubeadm, thanks.

File: pkg/kubectl/cmd/auth/BUILD [hunk 2]
```diff
@@ -34,6 +33,7 @@ go_library(
         "//staging/src/k8s.io/client-go/kubernetes/typed/authorization/v1:go_default_library",
         "//staging/src/k8s.io/client-go/kubernetes/typed/core/v1:go_default_library",
         "//staging/src/k8s.io/client-go/kubernetes/typed/rbac/v1:go_default_library",
+        "//staging/src/k8s.io/component-helpers/auth/rbac/reconciliation:go_default_library",
```
Comment (liggitt) [inline pkg/kubectl/cmd/auth/BUILD:36]: very nice. drop these from `pkg/kubectl/.import-restrictions`:

```
      - k8s.io/kubernetes/pkg/registry/rbac/reconciliation
      - k8s.io/kubernetes/pkg/registry/rbac/validation
```

**Comment by liggitt** [comment]
This looks way better, thanks. one comment on ratcheting the import restrictions, then the changes lgtm. Do each of the intermediate commits compile or should this be squashed?

File: pkg/kubectl/cmd/auth/BUILD (hunk 2 above)
Comment (ingvagabund) [reply pkg/kubectl/cmd/auth/BUILD:36]: Done

**Comment by ingvagabund** [comment]
> This looks way better, thanks. one comment on ratcheting the import restrictions, then the changes lgtm. Do each of the intermediate commits compile or should this be squashed?

Only some intermediate commits can be compiled. I kept them this way for easier review. Squashed now.

**Pull request merged**